leaf-ui/js/onFunctions.js

```diff
@@ -1151,67 +1176,34 @@ function removeHighlight(){
     }
 }
 
-/**
- * Function for single click on cell
- * 
- */
-paper.on('cell:pointerup', function(cellView, evt) {
-	if(mode == "Modelling") {
-        // Link
-        if (cellView.model instanceof joint.dia.Link) {
-            var link = cellView.model;
-            basicActorLink(link);
-            // Element is selected
-        } else {
-
-            selection.reset();
-            selection.add(cellView.model);
-
-            // Remove highlight of other elements
-            removeHighlight();
-
-            // Highlight when cell is clicked
-            cellView.highlight();
-
-            currentHalo = createHalo(cellView);
-
-            embedBasicActor(cellView.model);
-
-            clearInspector();
-
-            if (cellView.model instanceof joint.shapes.basic.Actor) {
-            	actorInspector.render(cellView.model);
-			} else {
-                elementInspector.render(cellView.model);
-			}
-
-        }
-    }
-});
-
 /**
  * Embeds an element into an actor boundary
  *
  * @param {joint.dia.cell} cell
  */
 function embedBasicActor(cell) {
-	if (!(cell instanceof joint.shapes.basic.Actor)) {
-		var ActorsBelow = paper.findViewsFromPoint(cell.getBBox().center());
-
-		if (ActorsBelow.length) {
-			for (var i = 0; i < ActorsBelow.length; i++) {
-				var actorCell = ActorsBelow[i].model;
-				if (actorCell instanceof joint.shapes.basic.Actor) {
-					actorCell.embed(cell);
-					var nodeID = cell.attributes.nodeID;
-					var actorID = actorCell.attributes.nodeID
-					model.getIntentionByID(nodeID).nodeActorID = actorID;
-					model.getActorByID(actorID).intentionIDs.push(nodeID);
-				}
-			}
-		}
-	}
-
+    var ActorsBelow = paper.findViewsFromPoint(cell.getBBox().center());
+
+    // intention always returns with ActorsBelow
+    // so always at least one element in list
+    if (ActorsBelow.length > 1) {
```

Comment:
Optional: What if you took the return from paper.findViewsFromPoint() and just filtered it like paper.findViewsFromPoint().filter(view => view.model instanceof joint.shapes.basic.Actor))? Then you could check the list to see if it contained any elements and just iterate through those. And if not go into your else statement. I think it might be worth it just for readability but obviously up to you

Reply:
Love this! Great idea!